Assets/RGLUnityPlugin/Scripts/SceneManager.cs

```diff
@@ -435,5 +465,43 @@ private static IEnumerable<Renderer> GetUniqueRenderersInGameObjects(IEnumerable
             foreach (var smr in smrs) yield return smr;
             foreach (var mr in mrs) yield return mr;
         }
+
+        /// <summary>
+        /// Searches through new rglObjects for ones containing IntensityTexture component.
+        /// If present, check if the found texture was already sent to RGL. 
+        /// If not, send it and write its identifier to sharedTextures dictionary.
+        /// After that assign rglTexture to the proper rglObject.
+        /// </summary>
+        private static void AddTextures(IEnumerable<RGLObject> rglObjects)
+        {
+            foreach (var rglObject in rglObjects)
+            {                
+                var intensityTextureComponent = rglObject.RepresentedGO.GetComponent<IntensityTexture>();
+
+                if( intensityTextureComponent != null)
+                {
+                    if( intensityTextureComponent.texture != null)
+                    {
+                        string textureID = $"r#{intensityTextureComponent.texture.GetInstanceID()}";
+
+                        RGLTexture rglTextureToAdd = new RGLTexture();
+                    
+                        if(!sharedTextures.ContainsKey(textureID))
+                        {
+                            rglTextureToAdd = new RGLTexture(intensityTextureComponent.texture);
+                        }
+                    
+                        if (rglTextureToAdd.rglTexturePtr != IntPtr.Zero)
+                        {
+                            rglObject.SetIntensityTexture(rglTextureToAdd);
+                        }
+                        else
+                        {                         
+                            Debug.LogWarning($"RGL Cannot assign texture. Not created yet!");
+                        }   
```

Comment:
1. Are you sure that texture sharing works here?
2. I wouldn't check low-level conditions here (`rglTextureToAdd.rglTexturePtr != IntPtr.Zero`). Leave it to the RGLMeshObject wrappers.

Reply:
2. Fixed.

1. I gueeeess...? The number of textures in the scene seems correct, and entities are linked.
What is Your concern?

Reply:
1. I can't see a place where you are adding textures to the `sharedTextures` dictionary.

Reply:
LOL.
Nice catch, my bug.